### ISSUE TITLE
Enable filtering 

### ISSUE THREAD

**Issue by mattmaurano** [mauranolab/mapping#265]
Setting filterText.<fieldName> * for relevant bed tracks should enable the filter box

We could also add score-based filtering where appropriate

See http://genome.ucsc.edu/goldenPath/help/trackDb/trackDbHub.html#filter

Looks like it depends on a more recent version of daler/trackhub -- see #264

**Comment by andremrsantos** [comment]
A couple quick questions regarding this issue:
- Should we implement these for all bigBed tracks generated by `MakeTrackhub.py`?
- Should `filterText.*` be `filterText.Name` ?
- Are you thinking on including `scoreMax` and `scoreMin` filters? or `scoreFilter`?

**Comment by mattmaurano** [comment]
Yes, we should apply these to all bigBed with sufficient columns, unless you find a reason not to.

Yes, use filterText.Name. This ought to work for any bed4+ (though the [trackDb definition](http://genome.ucsc.edu/goldenPath/help/trackDb/trackDbHub.html#filter) says "Filters are not supported in bed3 or bed4 files, even bed 3+x. The file must be at least a bed5", I think that's mistaken)

Take a look at each track type to identify reasonable score Max/Min

**Comment by andremrsantos** [comment]
In `MakeTrackhub.py` there are three bigBed tracks:
- Hotspots (bigBed 4)
- Peak (bigBed 3)
- Genotype (bigBed 9)
Thus, I've included to their view definition `filterText.Name` [to Hotspots & Genotype] and `scoreMin/Max` [to Genotype]. I've setup scoreMin to 0 and max 200 for Genotype tracks, which seem an adequate range.

We could include `scoreMin/Max` to bigWig tracks [Coverage, Density & Cuts], but I believe `viewLimits` does the job in this context.

Is there something I'm missing here?

**Comment by mattmaurano** [comment]
I don't think filterText is relevant for any of those.

hotspots and Genotype should be filterable on score, but make sure not to suppress anything by default.

No need to do anything for the bigwig tracks.

Enabling filterText is more relevant for the Assembly tracks, which will be a second part of this issue, but I want to see how this looks first.

**Comment by andremrsantos** [comment]
- Hotspots are added as bigBed 4 and scoreMin/Max requires bed5+
- I've checked the score range for Genotypes and most entries are around 100-150 with small deviations. I've also counted how many genotypes features lay outside the range 0-200 on 50 random samples and found none outside the range.
- Do you want to try adding these parameters as I mentioned to test the future interface?

**Comment by mattmaurano** [comment]
I see, analysis.sh cuts the hotspot startchs to bed4 so there can be no filtering without regenerating. So score should only apply to Genotypes. Yes, please try.

**Comment by andremrsantos** [comment]
I've made a trial run including the parameters I specified above and got the following formatting for Genotypes:

![image](https://user-images.githubusercontent.com/589051/145874827-c472f34c-3544-4ded-8381-e73363f2c8aa.png)

I think scoreMin and scoreMax looks fine, but I am not sure `filterText` will work, since on UCSC page it uses the following syntax `filterText.Name *` instead of `filterText Name *`. If this is the case, I envision an patch for trackhub to make `filter` fields work.

**Comment by mattmaurano** [comment]
I don't think filterText will be super useful for Genotypes. I guess someone might occasionally filter on SNP name, but that seems pretty rare. How does your example above turn out in the browser?

We need two changes:
1) hotspots should be filterable on score, but make sure not to suppress anything by default.

2) Enabling filterText is more relevant for the Assembly tracks. I will do this and as a separate commit against this issue, but I want to see how the first bit looks first.

**Comment by andremrsantos** [comment]
Upon review of the issue:
- I've discover that `scoreMin/Max` doesn't add a score filter field (it instead set up the shading options), but the filter is implemented by the field `scoreFilter`.
- I've setup a trial run adding the `scoreFilter` option to Genotype to `0` and setup the filtering limits to `0:200` (`scoreFilterLimits` option).
- Including these options adds the filtering fields as seem in the figure below 
![image](https://user-images.githubusercontent.com/589051/151576293-bab7c18a-d768-4d85-800d-f023b9cf0d54.png)
- Although the text indicate a score range from 0 to 1000, it doesn't allow values above 200.
- I've try to set-up to filter a value range such as the example at https://genome.ucsc.edu/goldenPath/help/hubQuickStartFilter.html, but in our browser, and UCSC it always appear only the minimal score filter box (like the figure above), instead of the two boxes in their example (see below)
![image](https://user-images.githubusercontent.com/589051/151591185-c6c27893-14d0-4754-871b-2953746e8e72.png)
- Regarding Hotspots score filter, we are currently not able to implement it because the browser uses the bigBed file which is generated as a bed4 file in `analysis.sh` (see below) https://github.com/mauranolab/mapping/blob/9562ae7f29f57050ec17f9ca67173d604ee77840/dnase/analysis.sh#L622
- My trial run is available at: `https://cascade.isg.med.nyu.edu/~ribeia01/scratch/trackhub_dev/hub.txt`

**Comment by mattmaurano** [comment]
Andre, can you put together a PR?

**Comment by andremrsantos** [comment]
Sure, can you just confirm which features you want to include:

- genotypes score filter (which is kind of enabled as is)
- text filter for assemblies

**Comment by mattmaurano** [comment]
Yes might as well do both

Have you worked out what the genotype score in the hub derives from?

**Comment by andremrsantos** [comment]
Okay. The score on genotypes tracks correspond to VCF Qual column
